src/CircleEditor/CircleEditorDocument.ts

```diff
@@ -476,35 +476,51 @@ export class CircleEditorDocument
   public sendEncodingData(message: any) {
     let fbb = flexbuffers.builder();
     fbb.startMap();
-    for (const key in message.data) {
-      fbb.addKey(key);
-      const val = message.data[key][0];
-      const valType = message.data[key][1];
-      if (valType === "boolean") {
-        if (val === "true" || val === true) {
-          fbb.add(true);
-        } else if (val === "false" || val === false) {
-          fbb.add(false);
-        } else {
-          Balloon.error("'boolean' type must be 'true' or 'false'.", false);
-          return;
-        }
-      } else if (valType === "int") {
-        const guessType = this.guessExactType(val);
-        if (guessType === "float") {
-          Balloon.error("'int' type doesn't include decimal point.", false);
-          return;
-        } else if (
-          guessType === "error" ||
-          guessType === "not supported type"
-        ) {
-          Balloon.error("it's not a number", false);
-          return;
+
+    if ("data" in message) {
+      for (const key in message.data) {
+        if (Object.prototype.hasOwnProperty.call(message.data, key)) {
+          fbb.addKey(key);
```

Comment:
(optional)
Do you need to add key to fbb in case of `message.data[key].length >= 2`?

Reply:
I put it as a guard to prevent it from accessing unassigned array. It was a part of my trial to get rid of defect. I will try the test without it and see if I can remove it.